pkg/sidecar-controller/framework_test.go

```diff
@@ -667,6 +667,15 @@ func newContentArrayWithReadyToUse(contentName, boundToSnapshotUID, boundToSnaps
 	}
 }
 
+func newContentWithVolumeGroupSnapshotHandle(contentName, boundToSnapshotUID, boundToSnapshotName, snapshotHandle, volumeGroupSnapshotHandle, snapshotClassName, desiredSnapshotHandle,
+	volumeHandle string, deletionPolicy crdv1.DeletionPolicy, creationTime, size *int64, withFinalizer bool, deletionTime *metav1.Time) []*crdv1.VolumeSnapshotContent {
+	content := newContentArrayWithDeletionTimestamp(contentName, boundToSnapshotUID, boundToSnapshotName, snapshotHandle, snapshotClassName, desiredSnapshotHandle, volumeHandle, deletionPolicy, creationTime, size, withFinalizer, deletionTime)
+	for _, c := range content {
+		c.Status.VolumeGroupSnapshotHandle = &volumeGroupSnapshotHandle
+	}
+	return content
```

Comment:
nit: I personally do not like functions that take this many parmaeters. There are better ways of doing this. Such as - using the builder pattern or using parameter types.

Reply:
@gnufied Yes i agree that this is going to make the life harder (read/maintain) but that required more work outside the scope of this PR as lot of cleanup is required in this file, is it accepted to do this cleanup as a followup?

Reply:
Can you file an issue for this?

Reply:
filed one here https://github.com/kubernetes-csi/external-snapshotter/issues/1232